src/core/server/saved_objects/service/lib/search_dsl/query_params.ts

```diff
@@ -251,16 +241,133 @@ export function getQueryParams({
   };
 
   if (search) {
-    bool.must = [
-      {
-        simple_query_string: {
-          query: search,
-          ...getFieldsForTypes(types, searchFields, rootSearchFields),
-          ...(defaultSearchOperator ? { default_operator: defaultSearchOperator } : {}),
-        },
-      },
-    ];
+    const useMatchPhrasePrefix = shouldUseMatchPhrasePrefix(search);
+    const simpleQueryStringClause = getSimpleQueryStringClause({
+      search,
+      types,
+      searchFields,
+      rootSearchFields,
+      defaultSearchOperator,
+    });
+
+    if (useMatchPhrasePrefix) {
+      bool.should = [
+        simpleQueryStringClause,
+        ...getMatchPhrasePrefixClauses({ search, searchFields, types, registry }),
+      ];
+      bool.minimum_should_match = 1;
+    } else {
+      bool.must = [simpleQueryStringClause];
+    }
   }
 
   return { query: { bool } };
 }
+
+// we only want to add match_phrase_prefix clauses
+// if the search is a prefix search
+const shouldUseMatchPhrasePrefix = (search: string): boolean => {
+  return search.trim().endsWith('*');
+};
+
+const getMatchPhrasePrefixClauses = ({
+  search,
+  searchFields,
+  registry,
+  types,
+}: {
+  search: string;
+  searchFields?: string[];
+  types: string[];
+  registry: ISavedObjectTypeRegistry;
+}) => {
+  // need to remove the prefix search operator
+  const query = search.replace(/[*]$/, '');
+  const mppFields = getMatchPhrasePrefixFields({ searchFields, types, registry });
+  return mppFields.map(({ field, boost }) => {
+    return {
+      match_phrase_prefix: {
+        [field]: {
+          query,
+          boost,
+        },
+      },
+    };
+  });
```

Comment:
Don't really know how this API works, but does it matter that we're creating a query like:
```
"should": [
        {
          "simple_query_string": {
            "query": "my-vis*",
            "fields": [
              "visualization.title",
              "visualization.description"
            ],
            "default_operator": "OR"
          }
        },
        {
          "match_phrase_prefix": {
            "visualization.title": {
              "query": "my-vis",
              "boost": 1
            }
          }
        },
        {
          "match_phrase_prefix": {
            "visualization.description": {
              "query": "my-vis",
              "boost": 1
            }
          }
        }
]
```

I was expecting it to be:
```
"should": [
        {
          "simple_query_string": {
            "query": "my-vis*",
            "fields": [
              "visualization.title",
              "visualization.description"
            ],
            "default_operator": "OR"
          }
        },
        {
          "match_phrase_prefix": {
            "visualization.title": {
              "query": "my-vis",
              "boost": 1
            },
            "visualization.description": {
              "query": "my-vis",
              "boost": 1
            }
          }
        }
]
```

Reply:
I also was tbh. But `match_phrase` prefix does not work that way unfortunately. That can creates a lot of clauses in the bool query, but when I discussed it with the ES teams, they told me it should not impact performances in any significative way.